src/jshint.js

```diff
@@ -4732,14 +4732,15 @@ var JSHINT = (function() {
       name = tkn.value;
     }
 
-    if (props[name] && _.has(props, name)) {
+    if ((props[name] && _.has(props, name)) || (name === "__proto__" &&
+        Object.getPrototypeOf(props) !== Object.prototype)) { // __proto__ has been modified
       warning("W075", state.tokens.next, msg, name);
-    } else {
-      props[name] = {};
     }
 
-    props[name].basic = true;
-    props[name].basictkn = tkn;
+    props[name] = {
```

Comment:
was this change to fix anything? I guess its okay, since it is a new property we are now referring to.. have scoured the code to see if there is anything it could break and seems okay (accessors seem to be the only thing storing something else against the property and they are set elsewhere)